test/metabase/driver/sql_jdbc/connection_test.clj

```diff
@@ -633,3 +633,96 @@
+                (is (= original-details details)))
+              (assoc details :test-override true))
+            ;; Create a connection spec - this should trigger the override
+            (let [spec (sql-jdbc.conn/db->pooled-connection-spec db)]
+              (is (some? spec))
+              (testing "Override function was called"
+                (is @override-called?)))))
+
+        (testing "Connection works normally outside override scope"
+          (reset! override-called? false)
+          (let [spec (sql-jdbc.conn/db->pooled-connection-spec db)]
+            (is (some? spec))
+            (testing "Override function not called outside scope"
+              (is (not @override-called?)))))))))
+
+(deftest with-overridden-connection-details-nested-test
+  (testing "Nested overrides create separate pools with different composite keys"
+    (mt/test-drivers (mt/normal-drivers)
+      (let [db            (mt/db)
+            db-id         (u/the-id db)
+            outer-called? (atom false)
+            inner-called? (atom false)]
+        (sql-jdbc.conn/invalidate-pool-for-db! db)
+
+        (driver/with-overridden-connection-details db-id
+          (fn [details]
+            (reset! outer-called? true)
+            (assoc details :outer-override true))
+          (sql-jdbc.conn/db->pooled-connection-spec db)
+          (testing "Outer override was called"
+            (is @outer-called?))
+          (let [outer-pool-count (count (filter (fn [[[pool-db-id _] _]]
+                                                  (= pool-db-id db-id))
+                                                @@#'sql-jdbc.conn/database-id->connection-pool))]
+            (testing "One pool exists after outer scope creates it"
+              (is (= 1 outer-pool-count)))
+
+            (driver/with-overridden-connection-details db-id
+              (fn [details]
+                (reset! inner-called? true)
+                (assoc details :inner-override true))
+              (sql-jdbc.conn/db->pooled-connection-spec db)
+              (testing "Inner override was called"
+                (is @inner-called?))
+              (let [inner-pool-count (count (filter (fn [[[pool-db-id _] _]]
+                                                      (= pool-db-id db-id))
+                                                    @@#'sql-jdbc.conn/database-id->connection-pool))]
+                (testing "Two pools exist when nested scopes have different overrides"
+                  (is (= 2 inner-pool-count)))))))
+
+        (testing "Both pools persist after scope exits"
+          (let [final-pool-count (count (filter (fn [[[pool-db-id _] _]]
+                                                  (= pool-db-id db-id))
+                                                @@#'sql-jdbc.conn/database-id->connection-pool))]
+            (is (= 2 final-pool-count))))))))
+
+(deftest with-overridden-connection-details-persistence-test
```

Comment:
You mentioned the idea to apply TTL to the "overridden connection" pools.

Would you imagine this TTL applying to *all* pools?

If we imagine Workspaces being used at scale, I'm still worried about the peak amount of pools being saved, if every workspace has its own pool. 

wdyt of having a separate map with a fixed size (LRU) being used for caching overridden pools only?